### ISSUE TITLE
Falsely report `UnusedLocalVariable` on arrow-function parameters in global function and anonymous class methods

### ISSUE THREAD

**Issue by Floberrot** [phpmd/phpmd#1306]
- **PHPMD version:** 3.x.dev
- **PHP Version:** 8.4.21
- **Installation type:** composer
- **Operating System / Distribution & Version:** Linux

### Current Behavior
falsely reports `UnusedLocalVariable` on closure/arrow-function parameters inside global functions and anonymous class methods

### Expected Behavior
Do not report those type of errors in case params are used inside global functions and anonymous methods

### Steps To Reproduce:
Samples of code in error:
```php
 public static function createNewClass(): object
  {
      return new class {
          public function myMethod(string $param): string
          {
              // spotted as UnusedLocalVariable.
              return $param;
          }
      };
  }
```
OR
```php
// anonymous function
namespace App\Something {
    function myFunction(object $arr): void
    {
        $arr->map(fn (string $value) => $value);
    }
}
```
# Result:
```bash 
 83   | VIOLATION | Avoid unused local variables such as '$param'.
 177  | VIOLATION | Avoid unused local variables such as '$value'.
```
